cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkTransactional.java

```diff
@@ -313,6 +324,22 @@ public void onTransactionCompleted(boolean jobSucceeded, @Nullable TransactionFa
     }
 
     @Override
+    public <T extends Dataset> T getDataset(String namespace, String name, Map<String, String> arguments,
+                                            AccessType accessType) throws DatasetInstantiationException {
+      T dataset = datasetCache.getDataset(namespace, name, arguments, accessType);
```

Comment:
Yes it makes the code cleaner. But this will break all the existing code using dataset cache(add a null to the first argument). Does it worth to do so?